### DISCUSSION TITLE
Running sent_test_mail becomes stuck

### DISCUSSION THREAD

**Discussion by cptalpdeniz** [getredash/redash#6583] [Q&A] [answered]
I just configured the MAIL parameters in the env file and as per the guide ran the `docker-compose run --rm server manage send_test_mail` however it's just stuck. It's registering bunch of things and thats it, nothing is happening and it's stuck at Registering Google Hangouts Chat (lot of things we are not using), what's happening here?

![image](https://github.com/getredash/redash/assets/17345222/0d9b9f6b-e98e-484f-a1e0-2cdf23191b0a)

**Comment by masayuki038** [comment]
Hi @cptalpdeniz 

I tried this command on master branch. However, the result was different.
```
[masayuki@localhost redash]$ docker compose run --rm server manage send_test_mail
[+] Creating 2/0
 ? Container redash-postgres-1  Running                                                                                                                                                                                                                    0.0s
 ? Container redash-redis-1     Running                                                                                                                                                                                                                    0.0s
/usr/local/lib/python3.8/site-packages/qds_sdk/commands.py:1124: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if options.mode is "1":
/usr/local/lib/python3.8/site-packages/qds_sdk/commands.py:1137: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if options.db_update_mode is "updateonly":
/usr/local/lib/python3.8/site-packages/qds_sdk/commands.py:1524: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if (total is 0) or (downloaded == total):
[2023-11-08 13:35:50,943][PID:1][INFO][xmlschema] Resource 'XMLSchema.xsd' is already loaded
Usage: manage.py [OPTIONS] COMMAND [ARGS]...
Try 'manage.py --help' for help.

Error: No such command 'send_test_mail'.
```
And I found the issue regarding `send_test_mail`. (This result is also different...)
https://github.com/getredash/redash/issues/4171

I have some questions:
1. What is your redash version?
2. Which `guide` did you refer to?
3. Could you let us know your `env`? (Of course, please hide some passowrds)

**Reply by cptalpdeniz** [reply]
Hello, in order:

1. Version: 8.0.0+b32245 (a16f551e) 
2. https://redash.io/help/open-source/setup here at the end of Mail Configuration the command is given to test send_email
3. 
```PYTHONUNBUFFERED=0
REDASH_LOG_LEVEL-DEBUG
REDASH REDIS_URL=redis://redis:6379/0
POSTGRES_PASSWORD=
REDASH_COOKIE_SECRET=
REDASH_SECRET_KEY=
REDASH_DATABASE_URL=
REDASH_MAIL_SERVER-email-smtp.us-west-2.amazonaws.com
REDASH_MAIL_PORT=25
REDASH_MAIL_USE_TLS=true
REDASH_MAIL_USERNAME=
REDASH_MAIL_PASSWORD=
REDASH_MAIL_DEFAULT_SENDER=no-reply@chartedbi.com
REDASH_HOST=https://chartedbi.com
```

I installed and setup Redash using the DigitalOcean marketplace

**Reply by cptalpdeniz** [reply, marked as answer]
I have recreated all the Amazon SES credentials server etc, updated SSL/TLS libraries did apt-get update & upgrade and also finally added env_file entry to each section of component in the `docker-compose.yml` file. It works now.

**Comment by masayuki038** [comment]
@cptalpdeniz Thanks for your report!
I don't know why `send_test_mail` didn't work on my local env... I will investigate it.

Closing this discussion.

**Reply by masayuki038** [reply]
I confirmed that `docker compose run --rm server manage send-test-mail` works fine.
It looks like we should enter `send-test-mail` instead of `send_test_mail`.
